packages/ui-components/src/components/Editor/SqlEditor.tsx

```diff
@@ -0,0 +1,186 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useState } from 'react';
+import parser from 'js-sql-parser';
+import BaseAceEditor from 'react-ace';
+import styled from 'styled-components';
+import 'ace-builds/src-noconflict/mode-pgsql';
+import 'ace-builds/src-noconflict/theme-xcode';
+import 'ace-builds/src-noconflict/ext-language_tools';
+import type { Ace } from 'ace-builds';
+import { GREY_DE } from '../../../stories/story-utils/theme/colors';
```

Comment:
@alexd-bes Does `ui-components` have a `theme.ts` or _somewhere_ a colour palette is centrally defined? This is the best my IDE could find, and my manual digging was unfruitful too, but this feels dirty 😕

See also **line 30** where I use this import. I’ll leave another comment there to elaborate

Reply:
Nope, ui-components inherits the theme from the parent, so if you use any theme variables that MUI would have it will work (same as you'd use inside a FE package like tupaia-web)